app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,8 @@
+class ProjectsController < ApplicationController
+  def show
+    @project = Project.find(params[:id])
+    @challenge = Challenge.find(@project.challenge_id)
```

Comment:
Similarly here, only send project to the view. Your project knows which challenge it belongs to, so no need to send the challenge to the view too.